### ISSUE TITLE
Image component throws error about dirname

### ISSUE THREAD

**Issue by patrickconroy** [tinacms/tinacms#1738]
## Description
Hello. Trying to use the `image` field, when clicking on the button to "Drag 'n' drop some files here, or click to select files" I get an error in the console complaining that `T.dirname` is not a function.

### Steps to reproduce
```
import { useForm, usePlugin } from 'tinacms';

const Form = (initialValues = {}) => ({
    label: 'Test',
    fields: [
        {
            name: 'img',
            label: 'Image',
            component: 'image',
        },
    ],
    initialValues,
});

const Edit = () => {
    const [editableData, form] = useForm(Form());

    usePlugin(form);

    return '...';
};

export default Edit;
```

### Expected result
I'd expect to be able to upload images.

### Actual result
I can't upload anything because the page errors with 
```
Uncaught TypeError: T.dirname is not a function👷‍ This is a development only message. It will be removed in production builds. 
    at cn (webpack://node_modules/tinacms/build/index.js?:480:15)
```
(This happens when webpacking development and production modes) 

#### Tinacms Package Versions
```
├── UNMET PEER DEPENDENCY @tinacms/form-builder@^0.3.0
├─┬ @tinacms/styles@0.29.0
├─┬ react-tinacms-date@0.37.0
├─┬ react-tinacms-editor@0.37.0
├─┬ react-tinacms-inline@0.37.0
├─┬ UNMET PEER DEPENDENCY tinacms@0.37.0
│ ├── @tinacms/alerts@0.33.0
│ ├── @tinacms/core@0.33.0
│ ├─┬ @tinacms/fields@0.37.0
│ ├─┬ @tinacms/form-builder@0.34.0
│ ├─┬ @tinacms/forms@0.33.0
│ ├── @tinacms/icons@0.36.0
│ ├── @tinacms/react-alerts@0.36.0
│ ├─┬ @tinacms/react-core@0.34.0
│ ├─┬ @tinacms/react-forms@0.36.0
│ ├── @tinacms/react-modals@0.36.0
│ ├── @tinacms/react-screens@0.36.0
│ ├─┬ @tinacms/react-sidebar@0.36.1
│ ├─┬ @tinacms/react-toolbar@0.37.0
│ ├─┬ @tinacms/styles@0.33.0
├─┬ tinacms-condition-field@1.0.6
├─┬ tinacms-file-field@1.2.1
```

#### Tools

| Tool    | Version |
| ------- | ------- |
| browser |     Chrome    |
| os      |     Mac    |
| node    |    15.8     |
| npm     |    7.5     |

**Comment by fabarea** [comment]
I have the same problem when opening the media library

<img width="589" alt="image" src="https://user-images.githubusercontent.com/620730/118671326-a9e38f00-b7f7-11eb-9371-1b0698304808.png">

Here is the skeleton of my **custom** store

```
export default class MyMediaStore {
  accept = '*'
  ...
  async previewSrc(src) {
    return src
  }
  async list(): Promise<MediaList> {
    const userImages = await ApiClient.query({
      query: GET_USER_IMAGES
    })

    const items = userImages.data.blogPost.userImages as Media[]
    return {
      items,
      offset: 0,
      limit: 10,
      totalCount: items.length
    }
  }
  ...
}
```

Example of JSON returned by the API

```
{
        id: 'img-left',
        type: 'file',
        filename: '436e8794303fe4f28211f8b7af9609b0.png',
        directory: '/',
        previewSrc: '/436e8794303fe4f28211f8b7af9609b0.png'
}
```

Versions from package.json:

```
"tinacms": "^0.41"
"react-tinacms-editor": "^0.41",
"react-tinacms-inline": "^0.41",
```

**Comment by DirtyF** [comment]
@fabarea Can you share some context here, what media store are you using? Any public repo where we could reproduce the issue?

**Comment by fabarea** [comment]
@DirtyF Thanks for your help. I found the issue. In webpack, I had this configuration.

```
    resolve: {
      fallback: {
         path: false
      },
```

Without the `fallback` configuration, I was getting this error on build time

```
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "path": require.resolve("path-browserify") }'
	- install 'path-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "path": false }
```

After installing `path-browserify`  and changing the configuration, it was OK again. I wonder if this is something that could be changed in the tina to avoid such fallback.

```
    resolve: {
      fallback: {
         path: require.resolve('path-browserify') }
      },
```

@patrickconroy sorry if I "spoiled" your bug report with mine - it maybe different, although it looked similar at the first glance.

**Comment by arpi0003** [comment]
@fabarea @DirtyF I'm experiencing the same issue. I have the same `fallback` config you do in my webpack config:
```
resolve: {
      fallback: {
         path: require.resolve('path-browserify') }
      },
```
The crash is happening [here](https://github.com/tinacms/tinacms/blob/main/packages/%40tinacms/toolkit/src/components/media/breadcrumb.tsx#L32)
I started experiencing the issue after upgrading past Tina 0.5. Any ideas?

**Comment by arpi0003** [comment]
@DirtyF once we upgraded to the latest @tinacms/toolkit version (0.53.0) and added
```
plugins: [
    // fix "process is not defined" error:
    // (do "npm install process" before running the build)
    new webpack.ProvidePlugin({
      process: 'process/browser',
    }),
  ]
```
to our webpack5 config, we got past this issue. I just wanted to leave a trail in case someone else was having this issue with their custom webpack config.

**Comment by DirtyF** [comment]
Thanks for the feedback @arpi0003, much appreciated 🙏

**Comment by jeffsee55** [comment]
@arpi0003 the issue you're seeing was introduced by a new build script from us. You _should_ be able to update to TinaCMS 0.55.2 to fix this, and would then no longer need to do anything with your webpack config.

So in general, I think the initial ticket was created when we weren't polyfilling Node built-ins a while back. Then was likely fixed for a few months and we re-broke it last week (which we now have fixed for the last time [here](https://github.com/tinacms/tinacms/pull/2066) 🤞 ).

**Comment by arpi0003** [comment]
@jeffsee55 we upgraded to 0.55.2. I think the polyfill for `path` is working now as we don't need the `path-browersify` fallback, but we encountered another issue where `process` was not defined, so we had to add the polyfill manually as described in this [SO post](https://stackoverflow.com/questions/65018431/webpack-5-uncaught-referenceerror-process-is-not-defined). I'm not a Next.js user, but maybe if your version now uses webpack5 some of these default polyfills were removed.
